apps/web/src/components/tx-flow/flows/CreateNestedSafe/ReviewNestedSafe.tsx

```diff
@@ -0,0 +1,150 @@
+  const version = getLatestSafeVersion(chain)
+
+  const safeAccountConfig = useMemo(() => {
+    if (!chain || !nestedSafes) {
+      return
+    }
+
+    const undeployedSafe = createNewUndeployedSafeWithoutSalt(
+      version,
+      {
+        owners: [safeAddress],
+        threshold: 1,
+      },
+      chain,
+    )
+    const saltNonce = Date.now().toString()
+
+    return {
+      ...undeployedSafe,
+      saltNonce,
+    }
+  }, [chain, safeAddress, nestedSafes, version])
+
+  const [predictedSafeAddress] = useAsync(async () => {
+    if (provider && safeAccountConfig) {
+      return predictAddressBasedOnReplayData(safeAccountConfig, provider)
+    }
+  }, [provider, safeAccountConfig])
+
+  useEffect(() => {
+    if (!chain || !safeAccountConfig || !predictedSafeAddress) {
+      return
+    }
+
+    const deploymentTx = {
+      to: safeAccountConfig.factoryAddress,
+      data: encodeSafeCreationTx(safeAccountConfig, chain),
+      value: '0',
+    }
+
+    const fundingTxs: Array<MetaTransactionData> = []
+
+    for (const asset of params.assets) {
+      const token = balances.items.find((item) => {
+        return item.tokenInfo.address === asset[SetupNestedSafeFormAssetFields.tokenAddress]
+      })
+
+      if (token) {
+        fundingTxs.push(
+          createTokenTransferParams(
+            predictedSafeAddress,
+            asset[SetupNestedSafeFormAssetFields.amount],
+            token.tokenInfo.decimals,
+            token.tokenInfo.address,
+          ),
+        )
+      }
+    }
+
+    const createSafeTx = async (): Promise<SafeTransaction> => {
+      const isMultiSend = fundingTxs.length > 0
+      return isMultiSend ? createMultiSendCallOnlyTx([deploymentTx, ...fundingTxs]) : createTx(deploymentTx)
+    }
+
+    createSafeTx().then(setSafeTx).catch(setSafeTxError)
+  }, [chain, params.assets, safeAccountConfig, predictedSafeAddress, balances.items, setSafeTx, setSafeTxError])
+
+  const onSubmit = () => {
+    if (!predictedSafeAddress) {
+      return
+    }
+    dispatch(
+      upsertAddressBookEntries({
+        chainIds: [safe.chainId],
+        address: predictedSafeAddress,
+        name: params.name,
+      }),
+    )
+  }
+
+  return (
+    <SignOrExecuteForm onSubmit={onSubmit}>
+      {predictedSafeAddress && (
+        <Grid
+          container
+          sx={{
+            gap: 1,
+          }}
+        >
+          <Grid item md>
+            <Typography
+              variant="body2"
+              sx={{
+                color: 'text.secondary',
+              }}
+            >
+              Nested Safe
+            </Typography>
+          </Grid>
+
+          <Grid data-testid="beneficiary-address" item md={10}>
+            <EthHashInfo
+              name={params.name}
+              address={predictedSafeAddress}
+              shortAddress={false}
+              hasExplorer
+              showCopyButton
+              showAvatar={false}
+            />
+          </Grid>
+        </Grid>
```

Comment:
Our general approach for various "Review" components is to just create and pass a SafeTx to `SignOrExecuteForm` which in turn renders an appropriate ["confirmation view"](https://github.com/safe-global/safe-wallet-monorepo/blob/dev/apps/web/src/components/tx/confirmation-views/index.tsx) based on the transaction-preview payload. Please check if this works in this case, otherwise the other signers will not see this nice review component.

Reply:
Is there a defined pattern to pass arbitrary values to the the `ConfirmationView`? Here we need to show the predicted Safe address.

Reply:
No, the point is to not pass and parameters because the goal is to render the same view for creators and signers of a tx. So you can rely only on the decoded txData.

Reply:
Included your suggestion in 4b3cd6a762afbbce0d3bd543605654e8f8578b3f. The Safe address is now predicted from the decoded data.